Source/CarthageKit/Dependency.swift

```diff
@@ -4,14 +4,32 @@ import Tentacle
 
 /// Uniquely identifies a project that can be used as a dependency.
 public enum Dependency {
+	/// Uniquely identifies a Binary Spec's resolved URL and its description
+	public struct BinaryURL: CustomStringConvertible {
+		/// A Resolved URL
+		public let url: URL
+
+		/// An Optional custom description
+		public let overloadedDescription: String?
```

Comment:
I'm not wild about this name. Maybe something with `resolved` in it?

Reply:
In this case, I wasn't sure what to call it: (Likely only beneficial for relative paths 'short hand')

Alternatives:
customDescription
identifier
...

Reply:
I was unsure about this here?

In this case, this struct is for Binary URLs, but the only use case I need this override is for relative Binary Spec paths.

relative is a tricky thing to consider b/c most urls have a relative path, But this isn't necessary for all urls?

Reply:
How about `resolvedDescription`? Maybe we could make it non-optional and include the correct value for absolute URLs as well?